Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        const string appServicesPath =
+    "/System/Library/Frameworks/ApplicationServices.framework/Versions/Current/ApplicationServices";
+
+        // https://developer.apple.com/documentation/coregraphics/1454595-cgdisplaycreateimage
+        [DllImport(appServicesPath, EntryPoint = "CGDisplayCreateImage")]
+        static extern /* CGImageRef */ IntPtr CGDisplayCreateImage(int displayId);
+
+        static Task<ScreenshotResult> PlatformCaptureAsync()
+        {
+            using var pool = new NSAutoreleasePool();
+            var location = NSApplication.SharedApplication.KeyWindow.Frame.Location;
```

Comment:
Can the app call with without a window? Like the AppDelegate ctor? Maybe this should use `?.` just to be safe. 

Not sure what we want to do if there is no window or no screen... Maybe an InvalidOpEx with a "helpful" message.

Reply:
In my understanding Xamarin.macOS and Xamarin.Forms need a window. So I think this could not be null.

Reply:
I implement a fallback with `CurrentMouseLocation`.